hathor/transaction/block.py

```diff
@@ -106,64 +103,6 @@ def create_from_struct(cls, struct_bytes: bytes, storage: Optional['TransactionS
 
         return blc
 
-    def calculate_height(self) -> int:
-        """Return the height of the block, i.e., the number of blocks since genesis"""
-        if self.is_genesis:
-            return 0
-        assert self.storage is not None
-        parent_block = self.get_block_parent()
-        return parent_block.get_height() + 1
-
-    def calculate_min_height(self) -> int:
-        """The minimum height the next block needs to have, basically the maximum min-height of this block's parents.
-        """
-        assert self.storage is not None
-        # maximum min-height of any parent tx
-        min_height = 0
-        for tx_hash in self.get_tx_parents():
-            tx = self.storage.get_transaction(tx_hash)
-            tx_min_height = tx.get_metadata().min_height
-            min_height = max(min_height, not_none(tx_min_height))
-
-        return min_height
-
-    def get_feature_activation_bit_counts(self) -> list[int]:
```

Comment:
We could but I don't see how that is way more useful compared to simply using `block.static_metadata.feature_activation_bit_counts`. In the same vein of our discussion about properties in the previous PR, I think wrapping it in a method adds unnecessary overhead and may convey that this is something more complex than a simple property access, which it's not.

Note that this is the same discussion for the `get_height()` method below, but in that case I kept it since it's called in so many places that I didn't want to change in this PR. `feature_activation_bit_counts`, on the other hand, was not called in many places and I used it as a guinea pig to test the new structure.

About mutability, I agree, accessing the list directly could be problematic. But this is an issue that already exists in the current code. Maybe we could change this attribute to be a tuple instead, but that should be done in a separate PR.

Reply:
I'm ok to make it a tuple in a future PR. So the goal would be to remove the `Block.get_height()` method and the `TransactionMetadata.height()` property in the future, right? I mean, they were not removed in this PR because it would be too much.

Reply:
Exactly

Reply:
Thanks!